dlp/src/main/java/com/example/dlp/Inspect.java

```diff
@@ -57,234 +66,307 @@
+              .setInspectJob(inspectJobConfig)
+              .build();
+
+      DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+
+      System.out.println("Job created with ID:" + dlpJob.getName());
+
+      final SettableApiFuture<Boolean> done = SettableApiFuture.create();
 
-      // asynchronously submit an inspect operation
-      OperationFuture<InspectOperationResult, InspectOperationMetadata> responseFuture =
-          dlpServiceClient.createInspectOperationAsync(inspectConfig, storageConfig, outputConfig);
+      // setup a Pub/Sub subscriber to listen on the job completion status
+      Subscriber subscriber =
+          Subscriber.newBuilder(
+              ProjectSubscriptionName.of(projectId, subscriptionId),
+              (pubsubMessage, ackReplyConsumer) -> {
+                if (pubsubMessage.getAttributesCount() > 0
+                    && pubsubMessage.getAttributesMap().get("DlpJobName").equals(dlpJob.getName())) {
+                  // notify job completion
+                  done.set(true);
+                  ackReplyConsumer.ack();
+                }
+              })
+              .build();
+      subscriber.startAsync();
+
+      // wait for job completion
+      try{
+        done.get(1, TimeUnit.MINUTES);
+        Thread.sleep(500);
+      } catch (Exception e){
+        System.out.println("Unable to verify job completion.");
+      }
 
-      // ...
-      // block on response, returning job id of the operation
-      InspectOperationResult inspectOperationResult = responseFuture.get();
-      String resultName = inspectOperationResult.getName();
-      InspectResult inspectResult = dlpServiceClient.listInspectFindings(resultName).getResult();
+      DlpJob completedJob =
+          dlpServiceClient.getDlpJob(
+              GetDlpJobRequest.newBuilder().setName(dlpJob.getName()).build());
 
-      if (inspectResult.getFindingsCount() > 0) {
+      System.out.println("Job status: " + completedJob.getState());
+      InspectDataSourceDetails inspectDataSourceDetails = completedJob.getInspectDetails();
+      InspectDataSourceDetails.Result result = inspectDataSourceDetails.getResult();
+      if (result.getInfoTypeStatsCount() > 0) {
         System.out.println("Findings: ");
-        for (Finding finding : inspectResult.getFindingsList()) {
-          System.out.print("\tInfo type: " + finding.getInfoType().getName());
-          System.out.println("\tLikelihood: " + finding.getLikelihood());
+        for (InfoTypeStats infoTypeStat : result.getInfoTypeStatsList()) {
+          System.out.print("\tInfo type: " + infoTypeStat.getInfoType().getName());
+          System.out.println("\tCount: " + infoTypeStat.getCount());
         }
       } else {
         System.out.println("No findings.");
       }
-    } catch (Exception e) {
-      e.printStackTrace();
-      System.out.println("Error in inspectGCSFileAsync: " + e.getMessage());
     }
     // [END dlp_inspect_gcs]
```

Comment:
Shouldn't this be one line down ?? after final `}` since we aren't hiding a return to the test app.

Reply:
Done.